woocommerce/wc-admin
# Update/table component

Fixes #92 – This PR drastically updates the Table component, to add a new TableCard, Table, and TableSummary. TableCard works as a parent component for Table, TableSummary, and the Pagination of the tabular data.

[The README has been updated with details and the three components' props.](https://github.com/woocommerce/wc-admin/blob/update/table-component/client/components/table/README.md)

![screen shot 2018-07-30 at 3 45 37 pm](https://user-images.githubusercontent.com/541093/43419690-a6019d28-940f-11e8-873c-be1dc5abd415.png)

Currently the table header only handles the "simple case" in #94, a following PR will also add the filter/compare functionality to the TableCard header. The demo shows checkboxes, but the interactions are not finished yet.

Each report will be able to use TableCard, but will need to set up the `rows`, `headers`, and `summary` objects themselves. See [`getHeadersContent`](https://github.com/woocommerce/wc-admin/blob/update/table-component/client/analytics/report/revenue.js#L64), [`getRowsContent`](https://github.com/woocommerce/wc-admin/blob/update/table-component/client/analytics/report/revenue.js#L78), [`getSummaryContent`](https://github.com/woocommerce/wc-admin/blob/update/table-component/client/analytics/report/revenue.js#L138) for examples.

The column toggle needs a little more thought too, so the toggles aren't working yet, but I wanted this to get a few eyes on it before I keep going 👍 

To Test

- View the revenue report
- Make sure the table looks as expected
- Read over the readme, check that the components as described/used make sense for all the reports
